Testing/Tests/graveyard_removal_test.py

```diff
@@ -0,0 +1,60 @@
+"""
+This class ensures that GraveyardRemoval.py correctly removes the graveyard from an h5m file.
+"""
+
+import os
+import pymoab
+import sys
+
+sys.path.insert(1,"../../PythonTool")
+
+from GraveyardRemoval import get_sets_by_category
+from GraveyardRemoval import locate_graveyard
+from GraveyardRemoval import format_file_name
+
+
+"Initialize a PyMOAB core instance and load in the h5m data file."
+mb = pymoab.core.Core()
+mb.load_file("../SampleData/fng_zip.h5m")
+
+
+def test_get_sets_by_category():
+	# Ensure this function returns the correct number of entities with the specified tag.
+	group_categories = get_sets_by_category(mb, "Group")
+	assert len(group_categories) == 8
+
+
+def test_default_locate_graveyard():
+	# Ensure this function returns the correct list of entity sets without the graveyard volume.
+    groups_to_write = locate_graveyard(mb)
+    assert groups_to_write == [12682136550675320105, 12682136550675320106,
+							   12682136550675320107, 12682136550675320108,
+							   12682136550675320109, 12682136550675320110,
+							   12682136550675320111]
+
+
+def test_print_locate_graveyard(capsys):
+	# Ensure this function prints the correct entity handle for the graveyard volume.
+	groups_to_write = locate_graveyard(mb, True)
+	out, err = capsys.readouterr()
+	assert ("12682136550675320112" in out) == True
+
+
+def test_default_format_file_name():
+	# Ensure this function returns the correct default output file name.
+	output_name = format_file_name('fng_zip.h5m')
+	assert (output_name == 'fng_zip_no_grave.h5m') == True
+
+
+def test_option_format_file_name():
+	# Ensure this function returns the correct user input file name.
+	output_name = format_file_name('fng_zip.h5m', 'test_output.h5m')
+	assert (output_name == 'test_output.h5m') == True
+
+
+def test_graveyard_removal():
+	# Ensure that GraveyardRemoval correctly removes the graveyard from an h5m file.
+	os.system("python ../../PythonTool/GraveyardRemoval.py ../SampleData/fng_zip.h5m")
+	size = os.path.getsize("fng_zip_no_grave.h5m")
+	assert size == 53390316
```

Comment:
I wonder if you can also use `filecmp()` here with a reference version of the file?

Reply:
I considered this as well, but for some reason the file returned is always different! Even if I run GraveyardRemoval on the exact same h5m file two times, it will not generate an identical file in content, but it will remain the same size. My guess to why this happens is perhaps some sort of time stamp?

Reply:
That is strange, indeed!

It would be interesting to use `h5diff` (a tool for looking at the differences between 2 HDF5 files) to understand this better - a curiosity rather than a priority.